src/AutoRest.CSharp/Mgmt/Output/Resource.cs

```diff
@@ -18,8 +18,13 @@ namespace AutoRest.CSharp.Mgmt.Output
 {
     internal class Resource : MgmtTypeProvider
     {
+        protected static readonly string ResourcePosition = "resource";
+        protected static readonly string CollectionPosition = "collection";
+
         private static readonly HttpMethod[] MethodToExclude = new[] { HttpMethod.Put, HttpMethod.Get, HttpMethod.Delete, HttpMethod.Patch };
 
+        protected virtual string Position => ResourcePosition;
```

Comment:
Can we assign the value in constructor instead of making property virtual?

Reply:
Make sense